test/test_irods_http_api.py

```diff
@@ -3339,6 +3339,90 @@ def test_report_operation(self):
         self.assertIn('zones', zone_report)
         self.assertGreaterEqual(len(zone_report['zones']), 1)
 
+    def test_adding_removing_and_modifying_zones(self):
+        headers = {'Authorization': 'Bearer ' + self.rodsadmin_bearer_token}
+
+        # Add a remote zone to the local zone.
+        # The new zone will not have any connection information or anything else.
+        zone_name = 'other_zone'
+        r = requests.post(self.url_endpoint, headers=headers, data={
+            'op': 'add',
+            'name': zone_name
+        })
+        self.logger.debug(r.content)
+        self.assertEqual(r.status_code, 200)
+        self.assertEqual(r.json()['irods_response']['status_code'], 0)
+
+        try:
+            # Show the new zone exists by executing the stat operation on it.
+            r = requests.get(self.url_endpoint, headers=headers, params={
+                'op': 'stat',
+                'name': zone_name
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+
+            result = r.json()
+            self.assertEqual(result['irods_response']['status_code'], 0)
+            self.assertEqual(result['exists'], True)
+            self.assertEqual(result['info']['name'], zone_name)
+            self.assertEqual(result['info']['type'], 'remote')
+            self.assertEqual(result['info']['connection_info'], '')
+            self.assertEqual(result['info']['comment'], '')
+
+            # The properties to update.
+            property_map = [
+                ('name',            'other_zone_renamed'),
+                ('connection_info', 'example.org:1247'),
+                ('comment',         'updated comment'),
+                # The following lines are special. They modify the permissions of
+                # the remote zone for visibility purposes (i.e. strict ACLs). See
+                # the help text of "iadmin modzonecollacl" for more details.
+                #('acl',             'read:rods'),
+                #('acl',             'null:rods')
```

Comment:
Revisit this.

Reply:
After some experimentation with `iadmin modzonecollacl`, adding a dedicated operation for it makes the most sense.

Removing.

Reply:
Done.